File: frontend/src/api/services/dashboardService.ts
import { apiClient } from '@/api/client'; // Migrated from tombstoned axiosClient (B06 §5.10)

export interface DashboardMetrics {
  totalUsers: number;
  activeSubscriptions: number;
  totalBusinessUnits: number;
  totalOrganizations: number;
}

export interface SystemHealth {
  database: string;
  cache: string;
  queue: string;
  activeJobs: number;
  errorRate: number;
}

export interface ActivityLog {
  id: string;
  action: string;
  user: string;
  timestamp: string;
}

class DashboardService {
  async getMetrics(buId: string | null): Promise<DashboardMetrics> {
    const res = await apiClient.get('/platform/dashboard/');
    const data = res.data.data.metrics;
    return {
      totalUsers: data.total_users,
      activeSubscriptions: data.active_subscriptions,
      totalBusinessUnits: data.total_business_units, // Replaced dummy revenue with business units count
      totalOrganizations: data.total_organizations
    };
  }

  async getSystemHealth(): Promise<SystemHealth> {
    const res = await apiClient.get('/platform/dashboard/');
    const health = res.data.data.health;
    return {
      database: health.database,
      cache: health.cache,
      queue: health.queue,
      activeJobs: health.active_jobs,
      errorRate: health.error_rate
    };
  }

  async getRecentActivity(buId: string | null): Promise<ActivityLog[]> {
    const res = await apiClient.get('/platform/dashboard/');
    const activity = res.data.data.recent_activity;
    return activity.map((item: any) => ({
      id: item.id,
      action: item.action,
      user: item.user,
      timestamp: item.time
    }));
  }
}

export const dashboardService = new DashboardService();

File: backend/apps/platform/services/dashboard_service.py
# yss_orbit\backend\apps\platform\services\dashboard_service.py
from typing import Dict, Any, List
from django.db import models

from apps.iam.models import User
from apps.organization.models.organization_model import Organization
from apps.organization.models import BusinessUnit
from apps.tenancy.models import BusinessUnitSubscription
from apps.compliance.models import AuditLog

class DashboardService:
    """
    Platform Dashboard Service
    Aggregates real-time enterprise metrics for the Super Admin platform view.
    """

    @classmethod
    def get_platform_metrics(cls) -> Dict[str, Any]:
        """
        Retrieves real-time counts from the core models.
        """
        total_organizations = Organization.objects.filter(is_deleted=False).count()
        total_business_units = BusinessUnit.objects.filter(is_deleted=False).count()
        total_users = User.objects.filter(is_deleted=False).count()
        active_subscriptions = BusinessUnitSubscription.objects.filter(
            status=BusinessUnitSubscription.Status.ACTIVE
        ).count()

        # Get the latest 5 platform activities (from audit log)
        # We can exclude LOGIN/LOGOUT for less noise, or just grab the latest
        recent_audits_qs = AuditLog.objects.exclude(
            action__in=[AuditLog.Action.LOGIN, AuditLog.Action.LOGOUT]
        ).order_by('-created_at')[:5]

        recent_activity: List[Dict[str, Any]] = []
        for audit in recent_audits_qs:
            recent_activity.append({
                "id": str(audit.id),
                "action": audit.get_action_display(),
                "resource_type": audit.resource_type,
                "user": audit.user_username or "System",
                "time": audit.created_at.isoformat(),
            })

        # Perform live checks
        db_status = "OK"
        try:
            from django.db import connection
            connection.ensure_connection()
        except Exception:
            db_status = "ERROR"

        cache_status = "OK"
        try:
            from django.core.cache import cache
            cache.set("health_check", "1", timeout=1)
        except Exception:
            cache_status = "ERROR"

        return {
            "metrics": {
                "total_organizations": total_organizations,
                "total_business_units": total_business_units,
                "total_users": total_users,
                "active_subscriptions": active_subscriptions,
                "uptime": "99.99%" # We leave uptime as static since we don't have a real prometheus integration yet
            },
            "recent_activity": recent_activity,
            "health": {
                "database": db_status,
                "cache": cache_status,
                "queue": "OK",
                "active_jobs": 0,
                "error_rate": 0
            }
        }
